### PR TITLE
OSOE-464: Executing dotnet test with process timeout

### PR DESCRIPTION
**Pull request by dministro**
[OSOE-464](https://lombiq.atlassian.net/browse/OSOE-464)

The primary goal here is to detach the `stdout` of the `dotnet test` process to avoid the issue described [here](https://github.com/dotnet/sdk/issues/9452#issuecomment-1210449614), and at the same time, we have the possibility to control the process timeout. However the `blame-hang-timeout` does a very similar thing, as I have seen in the logs the `dotnet test` process is still stuck while the `Data collector 'Blame' message: All tests finished running, Sequence file will not be generated.` message was recorded, this means something is happening outside the `blame` controller scope.

## Summary by CodeRabbit

- **New Features**
  - Implemented a timeout for .NET test processes with the ability to collect diagnostics if a hang occurs.
  
- **Documentation**
  - Updated workflow files to document the new `dotnet-test-process-timeout` input parameter.

- **Chores**
  - Added steps for installing `dotnet-dump` and uploading diagnostic dumps for test hangs.

- **Bug Fixes**
  - Introduced a mechanism to handle test process execution and prevent indefinite hangs during automated testing.

[OSOE-464]: https://lombiq.atlassian.net/browse/OSOE-464?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Piedone** [comment]
@coderabbitai review

I'm testing the AI code review tool [CodeRabbit](https://coderabbit.ai/) here. Please bear with me. And if it comments some useful feedback, please address it.

**Comment by Piedone** [comment]
@coderabbitai review

Sorry about that.

**Comment by Piedone** [comment]
So, this is a speculative but not necessarily definitive fix, as well as a debugging aid if processes still hang, right?

**Comment by dministro** [comment]
> So, this is a speculative but not necessarily definitive fix, as well as a debugging aid if processes still hang, right?

After fixing some deadlocks, I didn't find locking or similar stuff in the memory dumps anymore, and the blame hang timeout doesn't help, only theories are left and yes this change creates memory dumps for the processes in the process tree to help debug if the issue still remains. We will see it.

**Comment by Piedone** [comment]
OK, thanks.

File: .github/actions/test-dotnet/Invoke-SolutionTests.ps1 [hunk 1]
```diff
@@ -10,22 +10,25 @@ param ($Solution, $Verbosity, $Filter, $Configuration, $BlameHangTimeout)
...
 
     $Env:Lombiq_Tests_UI__DockerConfiguration__ContainerName = 'uitt-sqlserver'
 }
 
-$Env:Lombiq_Tests_UI__SqlServerDatabaseConfiguration__ConnectionStringTemplate = $connectionStringStem + $connectionStringSuffix
+$connectionString = @(
+    'Server=.;Database=LombiqUITestingToolbox_{{id}}'
+    $connectionSecurity
+    'MultipleActiveResultSets=True;Connection Timeout=60;ConnectRetryCount=15;ConnectRetryInterval=5;Encrypt=false'
+    'TrustServerCertificate=true'
```
Comment (sarahelsaig) [inline .github/actions/test-dotnet/Invoke-SolutionTests.ps1]: It seems quite arbitrary where you have semicolon and where it's a new string (the `TrustServerCertificate=true` on its own at the end). Why not break each component up?
```suggestion
    'Server=.'
    'Database=LombiqUITestingToolbox_{{id}}'
    $connectionSecurity
    'MultipleActiveResultSets=True'
    'Connection Timeout=60'
    'ConnectRetryCount=15'
    'ConnectRetryInterval=5'
    'Encrypt=false'
    'TrustServerCertificate=true'
```

**Review by sarahelsaig — CHANGES_REQUESTED** [review]

File: .github/actions/test-dotnet/Invoke-SolutionTests.ps1 (hunk 1 above)
Comment (dministro) [reply .github/actions/test-dotnet/Invoke-SolutionTests.ps1]: A bit of laziness 😶‍🌫️

**Pull request merged**